PWGCF/EbyEFluctuations/Tasks/netprotonCumulantsMc.cxx

```diff
@@ -851,28 +891,41 @@ struct NetprotonCumulantsMc {
   template <typename T>
   float getEfficiency(const T& candidate)
   {
-    // Find the pt bin index based on the track's pt value
-    int binIndex = -1;
-    // Get the array from the Configurable
-    auto ptBins = (std::vector<float>)cfgPtBins;
-    auto effProt = (std::vector<float>)cfgProtonEff;
-    auto effAntiprot = (std::vector<float>)cfgAntiprotonEff;
-
-    for (int i = 0; i < 16; ++i) {
-      if (candidate.pt() >= ptBins[i] && candidate.pt() < ptBins[i + 1]) {
-        binIndex = i;
-        break;
+    // Load eff from histograms in CCDB
+    if (cfgLoadEff) {
+      if (candidate.sign() > 0) {
+        float effmeanval = hRatio2DEtaVsPtProton->GetBinContent(hRatio2DEtaVsPtProton->FindBin(candidate.pt(), candidate.eta()));
+        return effmeanval;
       }
+      if (candidate.sign() < 0) {
+        float effmeanval = hRatio2DEtaVsPtAntiproton->GetBinContent(hRatio2DEtaVsPtAntiproton->FindBin(candidate.pt(), candidate.eta()));
+        return effmeanval;
+      }
+      return 0.0;
+    } else {
+      // Find the pt bin index based on the track's pt value
+      int binIndex = -1;
+      // Get the array from the Configurable
+      auto ptBins = (std::vector<float>)cfgPtBins;
+      auto effProt = (std::vector<float>)cfgProtonEff;
+      auto effAntiprot = (std::vector<float>)cfgAntiprotonEff;
```

Comment:
You don't need neither the casts nor new variables. Just use, for instance, `cfgPtBins.value[i]`